File: operacionesNumerosComplejos/numerosComplejos.dart
void main() {
  Complejo c1 = Complejo(1, 2);
  Complejo c2 = Complejo(3, 4);
  String stringc1 = c1.imprimir();
  String stringc2 = c2.imprimir();

  Complejo suma = c1.sumar(c2);
  Complejo resta = c1.restar(c2);
  Complejo multiplicacion = c1.multiplicar(c2);

  print('Numeros Complejos en Dart\n');
  print(stringc1);
  print(stringc2);
  print('suma = ${suma.real} + ${suma.imaginario}i');

  print('resta = ${resta.real} + ${resta.imaginario}i');

  print(
      'multiplicacion= ${multiplicacion.real} + ${multiplicacion.imaginario}i');
}

class Complejo {
  int real;
  int imaginario;

  Complejo(this.real, this.imaginario);

  Complejo sumar(Complejo numero) {
    return Complejo(
        this.real + numero.real, this.imaginario + numero.imaginario);
  }

  Complejo restar(Complejo numero) {
    return Complejo(
        this.real - numero.real, this.imaginario - numero.imaginario);
  }

  Complejo multiplicar(Complejo numero) {
    return Complejo(
        this.real * numero.real - this.imaginario * numero.imaginario,
        this.real * numero.imaginario + this.imaginario * numero.real);
  }

  String imprimir() {
    if (this.imaginario > 0)
      return '${real} + ${imaginario}i';
    else
      return '$real - ${imaginario}i';
  }
}

File: lab2_numerosComplejos/numerosComplejos.js
class Complejo {
    constructor(real, imaginaria) {
        this.real = real
        this.imaginaria = imaginaria
    }

    sumar(numero) {
        return new Complejo(this.real + numero.real, this.imaginaria + numero.imaginaria)
    }

    restar(numero) {
        return new Complejo(this.real - numero.real, this.imaginaria - numero.imaginaria)
    }

    multiplicar(numero) {
        return new Complejo((this.real * numero.real - this.imaginaria * numero.imaginaria), this.real * numero.imaginaria + this.imaginaria * numero.real)
    }

    imprimir() {
        if (this.imaginaria > 0)
            return console.log(this.real + ' + ' + this.imaginaria + 'i')
        else
            return console.log(this.real + ' - ' + this.imaginaria + 'i')
    }


}

console.log('Numeros Complejos en JavaScript\n')
let c1 = new Complejo(3, 2)
let c2 = new Complejo(1, 4)
let stringc1 = c1.imprimir(), stringc2 = c2.imprimir()

let suma = c1.sumar(c2)
let resta = c1.restar(c2)
let multiplicacion = c1.multiplicar(c2)

stringc1
stringc2
console.log('suma = ' + suma.real + ' + ' + suma.imaginaria + 'i')
console.log('resta = ' + resta.real + ' + ' + resta.imaginaria + 'i')
console.log('multiplicacion = ' + multiplicacion.real + ' + ' + multiplicacion.imaginaria + 'i')